### PR TITLE
Add upgrade thresholds to action thresholds

### LINKED ISSUE #4167: extend action thresholds to support package upgrade
currently action thresholds include key management and deploy. upgrade management action threshold will be added: Do the approvals have enough weight to upgrade the given package.

### PR DESCRIPTION
**Pull request by darthsiroftardis**
CHANGELOG:

- Add `upgrade_management` threshold to the action thresholds allowing users to set a threshold for addressable entities
- Allow migration path for legacy contracts to move into the new scheme
- Extended test suite to ensure migration logic of contracts installed in 1.5.x scheme moving into a 2.0 scheme
closes #4167

### CODE REVIEW

File: smart_contracts/contracts/explorer/faucet/README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ If you try to invoke the contract before these variables are set, then you'll ge
 
 | feature                  | cost             |
 |--------------------------|------------------|
-| faucet install           | `83_985_809_270` |
+| faucet install           | `84_007_229_270` |
 | faucet set variables     | `648_705_070`    |
-| faucet call by installer | `3_244_975_770`  |
+| faucet call by installer | `3_245_605_770`  |
```
Comment (EdHastingsCasperAssociation) [inline smart_contracts/contracts/explorer/faucet/README.md:40]: why does this get more expensive? should be orthogonal

File: execution_engine/src/runtime/mod.rs [hunk 2]
```diff
@@ -1685,9 +1695,13 @@ where
         bytes_written_ptr: u32,
         version_ptr: u32,
     ) -> Result<Result<(), ApiError>, Error> {
-        self.context.validate_key(&Key::from(package_hash))?;
+        // self.context.validate_key(&Key::from(package_hash))?;
+        //
+        // let mut package: Package = self.context.get_validated_package(package_hash)?;
```
Comment (EdHastingsCasperAssociation) [inline execution_engine/src/runtime/mod.rs:1700]: del

File: execution_engine/src/runtime/mod.rs [hunk 3]
```diff
@@ -1699,47 +1713,56 @@ where
...
             Some(previous_contract_hash) => {
-                let previous_contract: AddressableEntity =
+                let mut previous_entity: AddressableEntity =
                     self.context.read_gs_typed(&previous_contract_hash.into())?;
-                let previous_named_keys = previous_contract.named_keys().clone();
+
+                // Check if the calling entity must be grandfathered into the new
+                // addressable entity format
+                if previous_entity.associated_keys().is_empty()
+                    && self.context.validate_uref(&package.access_key()).is_ok()
```
Comment (EdHastingsCasperAssociation) [inline execution_engine/src/runtime/mod.rs:1725]: it's not clear to me why are we guarding on associated_keys being empty here?

**Comment by cspramit** [comment]
Try to review by 9th

**Comment by darthsiroftardis** [comment]
Closing in favor of #4320

**Pull request closed without merging**